### PR TITLE
PP-6388 Fix styling when showing Apple Pay and/or Google Pay buttons

### PR DESCRIPTION
**Pull request by iqbalgds**
- Update SCSS Styling to toggle when to show Apple/Google Pay headings & buttons
- Update charge.njk to use new styles

### CODE REVIEW

**Comment by heathd** [comment]
it would be helpful to include in the PR a description of the problem first, and then a description of what this PR does to fix it. As a reviewer, if I don't already have context on what the problem is, I have to try to reverse-engineer what the problem was, and then try to work out whether what you have done fixes it.

**Comment by iqbalgds** [comment]
This PR fixes the following styling issues:
# Problem 1.  
- Backend allows Apple & Google Pay
- Browser does NOT have Apple or Google Pay enabled
- `Charge` page is showing styling as though as though Apple or Google Pay is enabled.  All Apple & Google Pay styling should be hidden.
- Some of the Headings are at the wrong heading level - they are <h3> instead of <h2>.
- See Screenshot below.

# Problem 2
- Backend only allows Apple Pay.
- Browser has Apple Pay enabled
- `Charge` page is showing styling as though as if both Apple or Google Pay is enabled.  Styling should only be for Apple pay.
- Same problem happens if the backend only allows Google Pay and the Browser has Google Pay enabled.

File: app/assets/sass/modules/_web-payments.scss [hunk 1]
```diff
@@ -1,20 +1,35 @@
-#apple-pay-container,
-#google-pay-container {
+.apple-pay-container{
```
Comment (iqbalgds) [inline app/assets/sass/modules/_web-payments.scss:1]: Wraps the code for the Apple Pay Button code

File: app/assets/sass/modules/_web-payments.scss [hunk 2]
```diff
@@ -1,20 +1,35 @@
...
   display: none;
-}
 
-// Toggle Apple Pay
-.apple-pay-available {
-  #apple-pay-container {
+  .apple-pay-available & {
```
Comment (iqbalgds) [inline app/assets/sass/modules/_web-payments.scss:4]: Parent selector - only show this when a parent element has `apple-pay-available` on it

File: app/assets/sass/modules/_web-payments.scss [hunk 3]
```diff
@@ -1,20 +1,35 @@
...
     display: block;
   }
 }
 
-// Toggle Google Pay
-.google-pay-available {
-  #google-pay-container {
+.google-pay-container {
+  display: none;
```
Comment (iqbalgds) [inline app/assets/sass/modules/_web-payments.scss:10]: Same as above but for the Google Pay button.

File: app/assets/sass/modules/_web-payments.scss [hunk 4]
```diff
@@ -1,20 +1,35 @@
...
+
+  .google-pay-available & {
     display: block;
-  }
+  } 
+}
+
+.web-payment-button-heading{
+  display: none;
```
Comment (iqbalgds) [inline app/assets/sass/modules/_web-payments.scss:18]: This class is added to Headings that need to be displayed when either Apple or Google Pay button is available.
Hidden by default.

File: app/assets/sass/modules/_web-payments.scss [hunk 5]
```diff
@@ -1,20 +1,35 @@
...
+  
+  .apple-pay-available &,
+  .google-pay-available & {
+    display: block;
```
Comment (iqbalgds) [inline app/assets/sass/modules/_web-payments.scss:22]: Using a SCSS Parent selector - only show these headings when the body has `apple-pay-available` or `google-pay-available` added to it.
`apple-pay-available` and/or `google-pay-available` CSS classes are added by JS in the browser if the browser supports Apple Pay.

File: app/assets/sass/modules/_web-payments.scss [hunk 6]
```diff
@@ -1,20 +1,35 @@
...
+  
+  .apple-pay-available &,
+  .google-pay-available & {
+    display: block; 
+  } 
+}
+
+.non-web-payment-button-heading{
+  display: block;
+
```
Comment (iqbalgds) [inline app/assets/sass/modules/_web-payments.scss:28]: Opposite from above.  This class is added to Headings that need to be hidden when either Apple or Google Pay button is available.
Shown by default.

File: app/views/charge.njk [hunk 7]
```diff
@@ -63,60 +63,63 @@
               </div>
             </div>
             {% if allowApplePay or allowGooglePay %}
-              <h1 class="govuk-heading-l">{{ __p("cardDetails.enterPaymentDetails") }}</h1>
+              <h1 class="govuk-heading-l web-payment-button-heading">{{ __p("cardDetails.enterPaymentDetails") }}</h1>
```
Comment (iqbalgds) [inline app/views/charge.njk:66]: This heading needs to be hidden when there are no Apple/Google Pay buttons.
So add the `web-payment-button-heading` class to it.

File: app/views/charge.njk [hunk 8]
```diff
@@ -63,60 +63,63 @@
...
-              govukButton({
-                html: ApplePayButtonHTML,
-                classes: "web-payment-button web-payment-button--apple-pay",
-                attributes: {
-                  id: "apple-pay-payment-method-submit"
-                }
-              })
-            }}
-                <span id="apple-pay-payment-method-divider" class="pay-divider">
+              <div class="apple-pay-container">
```
Comment (iqbalgds) [inline app/views/charge.njk:69]: Wrapping all Apple Pay button code in a CSS container class..

File: app/views/charge.njk [hunk 9]
```diff
@@ -63,60 +63,63 @@
...
-              govukButton({
-                html: GooglePayButtonHTML,
-                classes: "web-payment-button web-payment-button--google-pay",
-                attributes: {
-                  id: "google-pay-payment-method-submit"
-                }
-              })
-            }}
-                <span id="google-pay-payment-method-divider" class="pay-divider">
+              <div class="google-pay-container">
```
Comment (iqbalgds) [inline app/views/charge.njk:93]: Wrapping all Google Pay button code in a CSS container class.

File: app/views/charge.njk [hunk 10]
```diff
@@ -63,60 +63,63 @@
...
+              </div>
             {% endif %}
             <div id="spinner" class="hidden"><img src="/images/spinner.gif" alt="{{ __p("authorisation.spinnerAltText") }}"/></div>
 
             <div id="card-details-wrap">
               {% if allowApplePay or allowGooglePay %}
-                <h2 class="govuk-heading-m">{{ __p("cardDetails.enterCardDetails") }}</h2>
-              {% else %}
-                <h1 class="govuk-heading-l">{{ __p("cardDetails.enterCardDetails") }}</h1>
+                <h2 class="govuk-heading-m web-payment-button-heading">{{ __p("cardDetails.enterCardDetails") }}</h2>
```
Comment (iqbalgds) [inline app/views/charge.njk:120]: This version of the heading only shows when there are Apple/Google Pay buttons.
So add the `web-payment-button-heading` class to it.

File: app/views/charge.njk [hunk 11]
```diff
@@ -63,60 +63,63 @@
...
               {% endif %}
+              <h1 class="govuk-heading-l non-web-payment-button-heading">{{ __p("cardDetails.enterCardDetails") }}</h1>
```
Comment (iqbalgds) [inline app/views/charge.njk:122]: This version of the heading only shows when there are Apple/Google Pay buttons.
So add the `non-web-payment-button-heading` class to it.
This is shown by default.

**Pull request merged**